src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1324,15 +1316,40 @@ .. chosenResolvedItem.Suppressions
                         continue;
                     }
 
+                    // See if a dependency with the same version and suppressions has already been chosen
+                    if (resolvedDependencyGraphItems.TryGetValue(childLibraryDependencyIndex, out ResolvedDependencyGraphItem? childResolvedDependencyGraphItem)
+                        && childResolvedDependencyGraphItem.LibraryRangeIndex == childLibraryRangeIndex
+                        && childResolvedDependencyGraphItem.Suppressions.Count == 1
+                        && childResolvedDependencyGraphItem.Suppressions[0].Count == 0
+                        && HasCommonAncestor(childResolvedDependencyGraphItem.Path, currentDependencyGraphItem.Path))
+                    {
+                        childResolvedDependencyGraphItem.Parents ??= new HashSet<LibraryRangeIndex>();
```

Comment:
I would prefer to see the instantiation of the HashSet encapsulated in the ResolvedDependencyGraphItem rather than having all consumers of this object be responsible for instantiating the Parents property HashSet on its behalf.